pymc3/distributions/multivariate.py

```diff
@@ -1637,27 +1637,21 @@ def random(self, point=None, size=None):
         mu, colchol, rowchol = draw_values(
             [self.mu, self.colchol_cov, self.rowchol_cov], point=point, size=size
         )
-        if size is None:
-            size = ()
-        if size in (None, ()):
-            standard_normal = np.random.standard_normal((self.shape[0], colchol.shape[-1]))
-            samples = mu + np.matmul(rowchol, np.matmul(standard_normal, colchol.T))
-        else:
-            samples = []
-            size = tuple(np.atleast_1d(size))
-            if mu.shape == tuple(self.shape):
-                for _ in range(np.prod(size)):
-                    standard_normal = np.random.standard_normal((self.shape[0], colchol.shape[-1]))
-                    samples.append(mu + np.matmul(rowchol, np.matmul(standard_normal, colchol.T)))
-            else:
-                for j in range(np.prod(size)):
-                    standard_normal = np.random.standard_normal(
-                        (self.shape[0], colchol[j].shape[-1])
-                    )
-                    samples.append(
-                        mu[j] + np.matmul(rowchol[j], np.matmul(standard_normal, colchol[j].T))
-                    )
-            samples = np.array(samples).reshape(size + tuple(self.shape))
+        size = to_tuple(size)
+        dist_shape = to_tuple(self.shape)
+        output_shape = size + dist_shape
+
+        # Broadcasting all parameters
+        mu = broadcast_dist_samples_to(to_shape=output_shape, samples=[mu], size=size)[0]
+        rowchol = np.broadcast_to(rowchol, shape=size + rowchol.shape[-2:])
+
+        colchol = np.broadcast_to(colchol, shape=size + colchol.shape[-2:])
+        perm = np.arange(len(output_shape))
+        perm[-2:] = [perm[-1], perm[-2]]
+        colchol = np.transpose(colchol, perm)
```

Comment:
I don't know how hard this would be, but it is possible to make a test case which would fail if this wasn't present?

Reply:
Good question. I would imagine to pass a custom `ref_random` (like one below) with transposed `covchol` and see if `stats.ks_2samp` results in failing tests.

https://github.com/pymc-devs/pymc3/blob/0402aab9594ec7c4f63c4ea35a52428d424652b4/pymc3/tests/test_distributions_random.py#L840:L842

Let me give it a try.

Reply:
Weird. Is this only at my side, that GitHub is unable to render permanlinks?

Reply:
I don't see it rendered either

Reply:
> I would imagine to pass a custom ref_random (like one below) with transposed covchol and see if stats.ks_2samp results in failing tests.

I added a test for the same in c33d571 .